src/Samples/Sarif.WorkItems.Sample/ProgramStubs.cs

```diff
@@ -0,0 +1,233 @@
+
+            var workItemFiler = new WorkItemFiler(
+                filingTarget,
+                filteringStrategy: new NewResultsFilteringStrategy(),
+                groupingStrategy: new OneResultPerWorkItemGroupingStrategy());
+
+            string logFilePath = Path.GetTempFileName();
+            File.WriteAllText(logFilePath, JsonConvert.SerializeObject(matchedLog));
+
+            // IMPORTANT! There's no facility defined yet for creating/configuring titles, area paths, etc.
+
+            try
+            {
+                IEnumerable<ResultGroup> resultGroups = workItemFiler.FileWorkItems(logFilePath).Result;
+                foreach (ResultGroup resultGroup in resultGroups)
+                {
+                    foreach (Result result in resultGroup.Results)
+                    {
+                        bool matched = guidToMatchedResultMap.TryGetValue(result.Guid, out Result originalResult);
+                        Debug.Assert(matched);
+                        originalResult.WorkItemUris = result.WorkItemUris;
+                    }
+                }
+            }
+            finally
+            {
+                if (File.Exists(logFilePath)) File.Delete(logFilePath);
+            }
+        }
+#endif
+
+        #region Stubs
+        private static SarifLog RetrieveCurrentSarif(string v)
+        {
+            string sarifText = GetResourceText("Sarif.Sdk.Sample.SampleTestFiles.Current.sarif");
+            return JsonConvert.DeserializeObject<SarifLog>(sarifText);
+        }
+
+        private static SarifLog RetrieveBaselineSarif(string v)
+        {
+            string[] resourceNames = typeof(Program).Assembly.GetManifestResourceNames();
+            Console.WriteLine(String.Join("\r\n", resourceNames));
+
+            string sarifText = GetResourceText("Sarif.Sdk.Sample.SampleTestFiles.Baseline.sarif");
+            return JsonConvert.DeserializeObject<SarifLog>(sarifText);
+        }
+
+        public static string GetResourceText(string resourceName)
+        {
+            string text = null;
+
+            using (Stream stream = typeof(Program).Assembly.GetManifestResourceStream(resourceName))
+            {
+                if (stream == null) { return null; }
+
+                using (StreamReader reader = new StreamReader(stream))
+                {
+                    text = reader.ReadToEnd();
+                }
+            }
+            return text;
+        }
+
+        private interface ISarifLogBaseliner
+        {
+            SarifLog CreateBaselinedSarifLog(SarifLog baselineSarif, SarifLog currentSarif, bool assignGuidsToNewResults);
+            SarifLog CreateBaselinedSarifLog(SarifLog baselineSarif, SarifLog currentSarif);
+        }
+
+        private class SarifLog
+        {
+            public IList<Run> Runs { get; internal set; }
+
+            internal static IEnumerable<WorkItemFilingMetadata> CreateWorkItemFilingMetadataFromLogs(IEnumerable<SarifLog> sarifLogsToFile)
```

Comment:
>CreateWorkItemFilingMetadataFromLogs [](start = 64, length = 36)

So this is called as `SarifLog.CreateWorkItemFilingMetadataFromLogs(sarifLogsToFile)`. I think I like it better as an extension method on `IEnumerable<SarifLog>`, so it's called as `sarifLogsToFile.CreateWorkItemFilingMetadata()`.

For that matter I like it best of all as an extension method on a single `SarifLog`, so:
```
IEnumerable<WorkItemFilingMetadata> metadatas /* yeah, I know */ = sarifLogsToFile.Select(log => log.CreateWorkItemFilingMetadata());
```

Reply:
Yes, an extension method had occurred to me as well. The API usability issue for this is typically one of discovery. IOW, if we do create this method, it should live in the core namespace (so that it appears in IntelliSense in expected situations)..

---
In reply to: [307082555](https://github.com/microsoft/sarif-sdk/pull/1589#discussion_r307082555) [](ancestors = 307082555)